src/easy_config/__init__.py

```diff
@@ -159,11 +174,15 @@ def _read_dict(
         :param d: the input mapping of string configuration value names to their values
         :returns: a mapping from string configuration value names to their values
         """
-        return {
-            field.name: field.type(d[field.name])
-            for field in dataclasses.fields(cls)
-            if field.name in d
-        }
+        values = {}
+        for field in dataclasses.fields(cls):
+            try:
+                if field.name in d:
```

Comment:
I try and isolate as much as possible... can you put the if statement all the way around the try?

Reply:
No reason not to!